docs/CODE_STYLE_GUIDE.md

```diff
@@ -0,0 +1,53 @@
+# Code Style Guide
+
+Read the code, and follow the existing style as much as possible.
+
+## ClangFormat
+
+For C++, Java and Objective C/C++ code we use [clang-format](http://clang.llvm.org/docs/ClangFormat.html).
+
+Install `clang-format` 21.0+ and make sure it's in your PATH.
+
+- To manually format a file run `clang-format -i file`.
+- To format all files in the repository run `tools/unix/clang-format.sh`
+- To format changes added to commit run `git clang-format`
```

Comment:
Can we add this as pre-commit hook? So that formatting will be done on every commit automatically...